### DISCUSSION TITLE
Allow to source non-dmp files for format conversions in one fell swoop

### DISCUSSION THREAD

**Discussion by TheChatty** [tasmota/decode-config#53] [Show and tell]
### Is your feature request related to a problem? Please describe

I want to edit a config.dmp to publish it.

* works: decode-config.py -s config.dmp --backup-file config.json
* works: nano config.json
* doesn't work: decode-config.py -s config.json --backup-file edited.dmp

### Describe the solution you'd like

decode-config.py -s config.json --backup-file.dmp

### Describe alternatives you've considered

* works: decode-config.py -s config.dmp --backup-file config.json
* works: nano config.json
* probably works: decode-config.py -s dummydevice --restore-file config.json
* probably works: decode-config.py -s dummydevice --backup-file edited.dmp

### Additional context

File conversion is the first marketed [program feature](https://github.com/tasmota/decode-config#readme).

**Comment by TheChatty** [comment]
It works if I do:
* decode-config -s config.dmp --restore-file config.json
* decode-config -s config.dmp --backup-file edited.json

That was not obvious me. You might add this as an example.

**Comment by curzon01** [comment]
use the following to edit it

1. decode-config.py -s config.dmp --backup-file config.json
2. nano config.json
3. cp config.dmp edited.dmp
4. decode-config.py -s edited.dmp --restore-file config.json

step 3 can be skipped if it is allowed to overwrite the original file config.dmp

1. decode-config.py -s config.dmp --backup-file config.json
2. nano config.json
3. decode-config.py -s config.dmp --restore-file config.json

option `-s` for 'source' might be missleading:
- 'source' is a source device/file when using backup function
- 'source' is a destination device/file when using restore function

see https://github.com/tasmota/decode-config/raw/master/media/pic/decode-config.png
